internal/consumer/partitionConsumer.go

```diff
@@ -243,6 +243,13 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 			p.tally.Gauge(metrics.KafkaPartitionTimeLag).Update(float64(lag))
 			p.tally.Gauge(metrics.KafkaPartitionReadOffset).Update(float64(m.Offset))
 			p.tally.Counter(metrics.KafkaPartitionMessagesIn).Inc(1)
+			delay := p.topicPartition.RetryDelay
+			// check message lag and non-zero retry delay, the assignment of topic RetryDelay is at topic initialization stage
+			if delay > 0 && lag > 0 && lag < delay {
```

Comment:
Is it necessary to check delay > 0 and lag > 0? Based on the docs for `time.Sleep`, it looks like `Sleep` will handle negative and 0 values correctly. https://golang.org/pkg/time/#Sleep

Reply:
good catch, will simplify the logic.